pkg/controller/jobs/leaderworkerset/leaderworkerset_webhook.go

```diff
@@ -0,0 +1,162 @@
+	if err != nil {
+		return err
+	}
+	if suspend {
+		if lws.Spec.LeaderWorkerTemplate.LeaderTemplate != nil {
+			wh.podTemplateSpecDefault(lws.Spec.LeaderWorkerTemplate.LeaderTemplate)
+		}
+		wh.podTemplateSpecDefault(&lws.Spec.LeaderWorkerTemplate.WorkerTemplate)
+	}
+
+	return nil
+}
+
+func (wh *Webhook) podTemplateSpecDefault(podTemplateSpec *corev1.PodTemplateSpec) {
+	if podTemplateSpec.Annotations == nil {
+		podTemplateSpec.Annotations = make(map[string]string, 1)
+	}
+	podTemplateSpec.Annotations[podcontroller.SuspendedByParentAnnotation] = FrameworkName
+	podTemplateSpec.Annotations[podcontroller.GroupServingAnnotation] = "true"
+}
+
+// +kubebuilder:webhook:path=/validate-leaderworkerset-x-k8s-io-v1-leaderworkerset,mutating=false,failurePolicy=fail,sideEffects=None,groups="leaderworkerset.x-k8s.io",resources=leaderworkersets,verbs=create;update,versions=v1,name=vleaderworkerset.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomValidator = &Webhook{}
+
+var (
+	labelsPath         = field.NewPath("metadata", "labels")
+	queueNameLabelPath = labelsPath.Key(constants.QueueLabel)
+	startupPolicyPath  = field.NewPath("spec", "startupPolicy")
+)
+
+func (wh *Webhook) ValidateCreate(ctx context.Context, obj runtime.Object) (warnings admission.Warnings, err error) {
+	lws := fromObject(obj)
+
+	log := ctrl.LoggerFrom(ctx).WithName("leaderworkerset-webhook")
+	log.V(5).Info("Validating create")
+
+	allErrs := jobframework.ValidateQueueName(lws.Object())
+	allErrs = append(allErrs, validateStartupPolicy(lws)...)
+
+	return nil, allErrs.ToAggregate()
+}
+
+func (wh *Webhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (warnings admission.Warnings, err error) {
+	oldLeaderWorkerSet := fromObject(oldObj)
+	newLeaderWorkerSet := fromObject(newObj)
+
+	log := ctrl.LoggerFrom(ctx).WithName("leaderworkerset-webhook")
+	log.V(5).Info("Validating update")
+
+	allErrs := apivalidation.ValidateImmutableField(
+		jobframework.QueueNameForObject(newLeaderWorkerSet.Object()),
+		jobframework.QueueNameForObject(oldLeaderWorkerSet.Object()),
+		queueNameLabelPath,
+	)
+	allErrs = append(allErrs, validateStartupPolicy(newLeaderWorkerSet)...)
+
+	return warnings, allErrs.ToAggregate()
+}
+
+func (wh *Webhook) ValidateDelete(context.Context, runtime.Object) (warnings admission.Warnings, err error) {
+	return nil, nil
+}
+
+func GetWorkloadName(lws *leaderworkersetv1.LeaderWorkerSet, groupIndex string) (string, error) {
+	ownerName := lws.Name
+	if lws.Spec.LeaderWorkerTemplate.LeaderTemplate != nil {
+		leaderHash, err := utilpod.GenerateShape(lws.Spec.LeaderWorkerTemplate.LeaderTemplate.Spec)
+		if err != nil {
+			return "", err
+		}
+		ownerName = fmt.Sprintf("%s-%s", ownerName, leaderHash)
+	}
+	workerHash, err := utilpod.GenerateShape(lws.Spec.LeaderWorkerTemplate.WorkerTemplate.Spec)
+	if err != nil {
+		return "", err
+	}
+	ownerName = fmt.Sprintf("%s-%s", ownerName, workerHash)
```

Comment:
I synced with @mimowo, and we decided that the `PodSpec` should be immutable except for the `Image` field, which allows users to perform a `RollingUpdate` to the newest version.